core/trino-main/src/main/java/io/trino/sql/routine/SqlRoutineAnalyzer.java

```diff
@@ -223,40 +223,40 @@ private static void validateArguments(FunctionSpecification function)
         }
     }
 
-    private static Optional<String> getLanguage(FunctionSpecification function)
+    private static Optional<Identifier> getLanguage(FunctionSpecification function)
     {
         List<LanguageCharacteristic> language = function.getRoutineCharacteristics().stream()
                 .filter(LanguageCharacteristic.class::isInstance)
                 .map(LanguageCharacteristic.class::cast)
-                .collect(toImmutableList());
+                .toList();
 
         if (language.size() > 1) {
-            throw semanticException(SYNTAX_ERROR, function, "Multiple language clauses specified");
+            throw semanticException(SYNTAX_ERROR, language.getLast(), "Multiple language clauses specified");
```

Comment:
error location at the second one would be even better

Reply:
it also doesn't cost more to call `.get(1)`

Reply:
Good point, fixed